syntax = "proto2";

enum PoplarOp {
  AvgPool = 0;
  AvgPoolGrad = 1;
  GroupNormGrad = 2;
  GroupNormInference = 3;
  GroupNormStatistics = 4;
  GroupNormTraining = 5;
  LstmLayerBwd = 6;
  LstmLayerFwd = 7;
  MaxPool = 8;
  MaxPoolGrad = 9;
  TruncatedNormal = 10;
  Relu = 11;
  ReluGrad = 12;
  TanhGrad = 13;
  Sigmoid = 14;
  SigmoidGrad = 15;
  Remap = 16;
  StatelessRandomUniform = 17;
  StatelessRandomUniformInt = 18;
  StatelessRandomNormal = 19;
  StatelessTruncatedNormal = 20;
  IpuInterCopy = 21;
  Dropout = 22;
  OneHot = 23;
  ReplicationFactor = 24;
  ReplicationNormalise = 25;
  ArgMin = 26;
  ArgMax = 27;
  PrintTensor = 28;
  StatefulGradientAccumulate = 29;
  StatefulGradientAccumulateAndAllReduce = 30;
  SelectScalarFromRows = 31;
  UpdateScalarInRows = 32;
  Fifo = 33;
  TopK = 34;
  UserOp = 35;
  RemapDeduce = 36;
  GRULayerBwd = 37;
  GRULayerFwd = 38;
  ReplicationIndex = 39;
  CodeletExpressionOp = 40;
  StatefulNoop = 41;
  CalcGfloatParams = 42;
  CastNativeToGfloat = 43;
  CastGfloatToNative = 44;
  MultiSlice = 45;
  MultiUpdate = 46;
  MultiUpdateAdd = 47;
  CreateBuffer = 48;
  AllGather = 49;
  SuggestRecompute = 50;
  BlockRecompute = 51;
  ReduceScatter = 52;
  Seed = 53;
  BufferLoadSlice = 54;
  BufferStoreSlice = 55;
  HostEmbeddingNotify = 56;
  HostEmbeddingLookup = 57;
  HostEmbeddingUpdate = 58;
  StatefulGradientAccumulateWithMomentum = 59;
  StatefulGradientAccumulateWithMomentumAndAllReduceWithNorm = 60;
  Gelu = 61;
  GeluGrad = 62;
  MultiConv = 63;
  SendRecvBarrier = 64;
  RemoteParameterLoad = 65;
  RemoteParameterStore = 66;
  // retired = 67;
  RecvFromHost = 68;
  SendToHost = 69;
  SliceApply = 70;
  SliceApplyaXbY = 71;
  SliceApplyabY = 72;
  SliceApplyaXb = 73;
  GradientAccumulatorCreate = 74;
  GradientAccumulatorAdd = 75;
  GradientAccumulatorSink = 76;
  WeightsTransposeChansFlipXY = 77;
  ExecutionCounter = 78;

  // Supported Fusions.
  Conv_biasadd = 200;
  Matmul_biasadd = 201;
  // retired = 202;
  Norm_scale_add = 203;
  Uniform_scale_add = 204;
  Wide_const = 205;
  Depthwise_conv = 206;
  Conv_with_reverse = 207;
  Bias_apply = 208;
  Zero_pad = 209;
  Depthwise_filter = 210;
  Scaled_inplace_aby = 211;
  Scaled_inplace_axby = 212;
  Conv_scaled_inplace = 213;
  Padding_reduce_window = 214;
  Implicit_binary = 215;
  Implicit_binary_inplace = 216;
  Implicit_ternary = 217;
  Implicit_ternary_inplace = 218;
  Scaled_inplace_axb = 219;
  Reduction_fp16_input = 220;
}
